controllers/upgrade/premisesChecker_test.go

```diff
@@ -0,0 +1,218 @@
+
+		// then
+		require.NoError(t, err)
+	})
+	t.Run("should fail for different dogu namespaces", func(t *testing.T) {
+		localDogu := readTestDataLdapDogu(t)
+		remoteDogu := readTestDataLdapDogu(t)
+		remoteDogu.Name = remoteDogu.GetNamespace() + "/test"
+		// when
+		err := checkDoguIdentity(localDogu, remoteDogu, false)
+
+		// then
+		require.Error(t, err)
+		assert.Equal(t, "dogus must have the same name (ldap=test)", err.Error())
+	})
+	t.Run("should fail for different dogu names", func(t *testing.T) {
+		localDogu := readTestDataLdapDogu(t)
+		remoteDogu := readTestDataLdapDogu(t)
+		remoteDogu.Name = "different-ns/" + remoteDogu.GetSimpleName()
+		// when
+		err := checkDoguIdentity(localDogu, remoteDogu, false)
+
+		// then
+		require.Error(t, err)
+		assert.Equal(t, "dogus must have the same namespace (official=different-ns)", err.Error())
+	})
+}
+
+func Test_premisesChecker_Check(t *testing.T) {
+	ctx := context.TODO()
+
+	t.Run("should succeed", func(t *testing.T) {
+		fromDoguResource := readTestDataRedmineCr(t)
+		fromDogu := readTestDataDogu(t, redmineBytes)
+		toDogu := readTestDataDogu(t, redmineBytes)
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(nil)
+		mockedChecker.On("ValidateDependencies", fromDogu).Return(nil)
+		mockedChecker.On("CheckDependenciesRecursive", fromDogu, fromDoguResource.Namespace).Return(nil)
+
+		sut := NewPremisesChecker(mockedChecker, mockedChecker, mockedChecker)
+
+		// when
+		err := sut.Check(ctx, fromDoguResource, fromDogu, toDogu)
+
+		// then
+		require.NoError(t, err)
+		mockedChecker.AssertExpectations(t)
+	})
+	t.Run("should fail when dogu identity check fails", func(t *testing.T) {
+		fromDoguResource := readTestDataRedmineCr(t)
+		fromDogu := readTestDataDogu(t, redmineBytes)
+		toDogu := readTestDataDogu(t, redmineBytes)
+		toDogu.Name = "somethingdifferent"
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(nil)
+		mockedChecker.On("ValidateDependencies", fromDogu).Return(nil)
+		mockedChecker.On("CheckDependenciesRecursive", fromDogu, fromDoguResource.Namespace).Return(nil)
+
+		sut := NewPremisesChecker(mockedChecker, mockedChecker, mockedChecker)
+
+		// when
+		err := sut.Check(ctx, fromDoguResource, fromDogu, toDogu)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "dogus must have the same name")
+		// there is no assert.IsNoType() assertion so we test it by negative type assertion
+		_, ok := err.(*requeueablePremisesError)
+		assert.False(t, ok)
+		mockedChecker.AssertExpectations(t)
+	})
+	t.Run("should fail when dependency validator fails", func(t *testing.T) {
+		fromDoguResource := readTestDataRedmineCr(t)
+		fromDogu := readTestDataDogu(t, redmineBytes)
+		toDogu := readTestDataDogu(t, redmineBytes)
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(errors.New("CheckWithResource"))
```

Comment:
You cloud use the test error object testing.AnError instead of creating a new error.